[29기 김지윤] Westagram 과제 로그인 페이지 ∙ 메인 페이지

memoyoon opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [X] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 아이디, 비밀번호 입력창을 사용한 로그인 기능 구현 
- 메인 페이지 html, css 구현 

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 로그인 입력 데이터 유효성 검사 
- 입력창에 변화가 일어날 때 마다 유효성 검사 함수가 실행된다.
- 유효성 검사 함수의 반환값(boolean)에 따라 버튼의 배경색이 변화한다.

2. 메인 페이지 
- html 태그 class이름을 지을 때 가독성이 좋고 유지보수가 비교적 쉽게 할 수 있도록 고민하며 class 이름을 지었습니다. 
- 오른쪽 메뉴 리스트에 overflow를 추가하여 스크롤 기능을 넣었습니다. 

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 사용자 관점에서 보기 좋은 웹사이트를 구현하고자 고민하였습니다. 

<br />

## :: 기타 질문 및 특이 사항
- 아직 js를 구현하지 못했습니다. 추후에 곧 완성하겠습니다.

dipsiiiiiiiiii commented:
우선 질문 주신 부분부터 답변 드리겠습니다!
질문 : 기존 HTML 마크업을 진행할 때, 해당 태그를 지칭하는 class를 필수적으로 붙였습니다. (ex_ <a class="forget_pw">)이유는 해당 클래스가 어떤 의미인지 명확히 구분하고 CSS를 구현할 때나 여러 페이지에 동일한 요소들이 들어있을때 선택자로 유용하게 쓰일 것 같기 때문이었습니다. 그런데 종종 시멘틱 태그들과 겹치는 부분이 있는 것 같습니다(ex_<main class="main_contents">) 앞으로 작업할 때 클래스명을 선언해주는 것이 불필요한 작업인 지에 대해 고민이 됩니다.
답변 : 좋은 고민을 하신 것 같아 좋네요! 클래스는 해당 태그에 스타일링을 할 때 사용하지만 클래스를 준 태그 바로 아래의 태그들에 스타일링을 줄 때는 굳이 클래스를 부여 할 필요가 없는 경우도 있습니다. 진영님이 지금 하신 것처럼 모든 태그가 클래스를 가질 필요는 없지만 가지고 있더라도 문제될 것은 없습니다!
앞으로 sass, styled-components 등 진영님이 고민하신 클래스에 대한 고민을 해결해 줄 수 있는 라이브러리를 배우고 사용할 예정이어서 기대 많이 해주세요 :)

질문 : 메인페이지 좋아요 버튼의 경우, 동일한 버튼이 여러개 있습니다. 이 때, for구문으로 구현하는 것이 나을까요? onclick을 사용하거나 다른 방법이 있는 지 궁금합니다.
답변 : 이 부분은 리뷰 하면서 설명해드릴게요!

질문 : 댓글기능을 추가하면서 발견한 문제인데, 기존에 있던 댓글 / 추가된 댓글 스타일에 차이가 있습니다. 기존댓글은 id / 댓글내용 사이에   만큼의 영역이 잡혀있는데 새로 추가한 댓글들은 id/댓글내용 사이 여백이 없습니다. 현재 css margin으로 여백을 주어 임의로 해결한상태인데, js로 추가한 댓글과 기존댓글 여백차이는 왜 생기는 걸까요?

답변 : 아직 코드를 보지 않았지만 제가 봤을 때는 태그가 기본적으로 가지고 있는 스타일링 때문인 것 같습니다. 그래서 프로젝트를 시작할 때 할상 reset.css 파일에서 모든 태그에 브라우저가 부여한 속성들을 초기화 하는 작업을 해주어야 합니다!

dipsiiiiiiiiii commented on students/16th/jinyounglim/login.html:
```diff
@@ -0,0 +1,25 @@
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>로그인 • westagram</title>
```
좋습니다~

dipsiiiiiiiiii commented on students/16th/jinyounglim/login.html:
```diff
@@ -0,0 +1,25 @@
+    <link rel="stylesheet" href="style/common.css">
+    <link rel="stylesheet" href="style/login.css">
+    <link rel="icon" sizes="192x192" href="https://www.instagram.com/static/images/ico/favicon-192.png/68d99ba29cc8.png">
+</head>
+<body>
+    <section class="login_page">
+        <div class="login_box">
+            <h1 class="logo">Westargram</h1>
+            <form class="login_form">
+                <input class="user_id" type="text" placeholder="전화번호, 사용자 이름 또는 이메일" value autofocus>
+                <input class="user_pw" type="password" placeholder="비밀번호" value>
+                <button class="btn_login">로그인</button>
+            </form>
+            <a class="forget_pw" href="https://www.instagram.com/accounts/password/reset" target="_blank">비밀번호를 잊으셨나요?</a>
+        </div>
+    </section>
+    <script src="js/login.js"></script>
+</body>
```
크! 세상 깔끔하게 태그 작성해주셨어요! LGTM👍

dipsiiiiiiiiii commented on students/16th/jinyounglim/main.html:
```diff
@@ -0,0 +1,155 @@
+    <link rel="icon" sizes="192x192" href="https://www.instagram.com/static/images/ico/favicon-192.png/68d99ba29cc8.png">
+</head>
+<body>
+    <header>
+        <nav>
```
header태그는 h1~h6 태그와 같이 사용합니다! 여기서는 nav 태그만 있으면 될 것 같네요!

dipsiiiiiiiiii commented on students/16th/jinyounglim/style/common.css:
```diff
@@ -0,0 +1,86 @@
+}
+
+body {
+    line-height: 1;
+    background-color: var(--bg-color);
+}
+
+ol, ul {
+	list-style: none;
+}
+
+blockquote, q {
+	quotes: none;
+}
+
+blockquote:before, blockquote:after,
+q:before, q:after {
+	content: '';
+	content: none;
+}
+
+table {
+	border-collapse: collapse;
+	border-spacing: 0;
+}
+
+button{
+	border: none;
+	background: none;
+	padding: 0;
+	cursor: pointer;
+}
+
+a{
+    text-decoration: none;
+}
+
+h1{
+	font-family:  'Lobster', cursive;
+}
```
이런 초기화 속성은 reset.css에 넣어주세요:)

dipsiiiiiiiiii commented on students/16th/jinyounglim/style/main.css:
```diff
@@ -0,0 +1,408 @@
+    background: url(../img/icons_sprit) no-repeat;
+    background-size: 85px;
+    background-position: -32px -23px;
+    text-indent: -9999px;
```
ㅋㅋㅋㅋㅋㅋㅋㅋㅋ

dipsiiiiiiiiii commented on students/16th/jinyounglim/style/common.css:
```diff
@@ -0,0 +1,86 @@
+@import url('https://fonts.googleapis.com/css2?family=Lobster&display=swap');
+:root {
+    --bg-color: #fafafa;
+    --border-gray: solid 1px #dbdbdb;
+    --id-color: #262626;
+    --sub-color: #8e8e8e;
+    --high-color: #0095f6;
+    --font-bold: 600;
+}
```
Good!!

dipsiiiiiiiiii commented on students/16th/jinyounglim/main.html:
```diff
@@ -0,0 +1,155 @@
+                        <strong class="user_id">delheure29</strong>
+                        위스타그램 첫 포스트! 앞으로 화이팅!
+                    </p>
+                    <button class="btn_txt_all">더보기</button>
```
더 보기 버튼이니까 btn_txt_all 보다는 more가 들어가는 게 더 좋겠네요!

dipsiiiiiiiiii commented on students/16th/jinyounglim/js/main.js:
```diff
@@ -0,0 +1,45 @@
+const btnTxtAll = document.querySelector('.btn_txt_all');
+const feedTxt = document.querySelector('.feeds_txt p');
+
+btnTxtAll.addEventListener('click', txtShow = () => {
```
이벤트에서 함수를 정의할 때는 익명함수로도 충분합니다!
```suggestion
btnTxtAll.addEventListener('click', () => {
```

dipsiiiiiiiiii commented on students/16th/jinyounglim/js/main.js:
```diff
@@ -0,0 +1,45 @@
+    });
+}
+
+const cmBox = document.querySelector('.comments_list');
```
```comment``` 줄임말은 ```cmt```가 조금 더 친절하겠네요!

dipsiiiiiiiiii commented on students/16th/jinyounglim/js/main.js:
```diff
@@ -0,0 +1,45 @@
+    btnTxtAll.remove();
+});
+
+const likes = document.querySelectorAll('.btn_heart');
+
+for(const like of likes){ 
+    like.addEventListener('click', function(){
+        this.classList.toggle('on');
+    });
+}
```
지금 이 로직으로 하게 되면 새로운 커멘트에는 이벤트가 부여되지 않고 있습니다.
따라서 새로운 댓글이 생성될 때 이벤트를 같이 부여한 다음에 ```comment_list```에 추가하면 되겠네요!

delheure replied:
기존 이벤트 likesActive 함수로 바꾸고 document.addEventListener('click', likesActive); 로 접근하려했는데 적용되지 않습니다ㅠㅠ 다른 방법이 있을까요?

dipsiiiiiiiiii commented on students/16th/jinyounglim/js/main.js:
```diff
@@ -0,0 +1,45 @@
+    cmTxt.innerHTML = cmInput.value;
+    id.innerHTML = 'delheure29';
+    
+    if(cmInput.value){
+        cmBox.appendChild(cm);
+        cm.appendChild(id);
+        cm.appendChild(cmTxt);
+        cm.appendChild(cmHeart);     
+        cmInput.value = "";
+    }
```
커멘트 인풋의 값이 있을 때만 이 함수를 실행하고 싶으신 거라서
함수 제일 위에서 아래 로직을 추가하면 위의 조건문은 필요 없을 것 같네요!
```
if(!cmInput.value){
  return;
}
```

dipsiiiiiiiiii commented on students/16th/jinyounglim/js/main.js:
```diff
@@ -0,0 +1,45 @@
+cmBtn.addEventListener('click', cmPut = (e) => {
+    e.preventDefault();
+
+    const cm = document.createElement('li');
+    const cmTxt = document.createElement('p');
+    const id = document.createElement('strong');
+
+    const cmHeart = document.createElement('button');
+    cmHeart.className ='btn_heart btn_small btn_com';
+    id.className = 'user_id';
+
+    cmTxt.innerHTML = cmInput.value;
+    id.innerHTML = 'delheure29';
```
innerHTML을 사용하면 하나하나 createElement 하지 않아도 됩니다! 한 번 검색해서 활용해보세요!

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다 진영님 :)
코드를 정말 깔끔하게 잘 짜시네요!
강아지들 너무 귀엽습니다 ㅎㅎ
리뷰 반영해주세요!

memoyoon committed:
깃허브, 깃 연습 파일입니다.

memoyoon committed:
Westagram 로그인 페이지 : js, css 구현 완료
Westagram 메인 페이지 : css 구현 완료

memoyoon committed:
westagram 로그인 페이지 ∙ 메인 페이지
1. 로그인 페이지
1) HTML, CSS 활용하여 로그인 페이지 레이아웃 구현
2) id, pw 입력 시 로그인 버튼 활성화 기능 구현
3) id에 '@' 포함한 문자 입력, pw 입력, 로그인 버튼 클릭 시 메인페이지로 이동
4) 'id, pw 힌트' alert창 구현

2. 메인 페이지
1) HTML, CSS 활용하여 메인 페이지 레이아웃 구현
2) 댓글 내용 입력 후 Enter press, 혹은 게시 버튼 클릭 시 댓글 추가 기능
3) 댓글 내용 입력 후 댓글 작성 시간 변경 ('35분 전' -> '방금 전')
4) 스토리∙회원님을 위한 추천 스크롤 기능 추가

memoyoon committed:
로그인 페이지
1) 아이디에 '@' 포함 되지 않을 경우 로그인 버튼 클릭 시 alert창 구현
